File: compiler/testData/loadJava/kotlinSignature/propagation/return/InheritNullabilityJavaSubtype.java
package test;

import org.jetbrains.annotations.NotNull;
import java.lang.CharSequence;

import jet.runtime.typeinfo.KotlinSignature;

public class InheritNullabilityJavaSubtype {
    @NotNull
    public CharSequence foo() {
        throw new UnsupportedOperationException();
    }

    public class Sub extends InheritNullabilityJavaSubtype {
        public String foo() {
            throw new UnsupportedOperationException();
        }
    }
}

File: compiler/testData/loadJava/kotlinSignature/propagation/return/InheritNullabilityJavaSubtype.kt
package test

import org.jetbrains.annotations.NotNull

public open class InheritNullabilityJavaSubtype : java.lang.Object() {
    public open fun foo(): CharSequence = throw UnsupportedOperationException()

    public open class Sub: InheritNullabilityJavaSubtype() {
        override fun foo(): String = throw UnsupportedOperationException()
    }
}